R/modules.R

```diff
@@ -31,11 +31,36 @@ createSessionProxy <- function(parentSession, ...) {
   # but not `session$userData <- TRUE`) from within a module
   # without any hacks (see PR #1732)
   if (identical(x[[name]], value)) return(x)
+
+  # Special case for $options (issue #3112)
+  if (name == "options") {
+    session <- find_ancestor_session(x)
+    session[[name]] <- value
+    return(x)
+  }
+
   stop("Attempted to assign value on session proxy.")
 }
 
 `[[<-.session_proxy` <- `$<-.session_proxy`
 
+# Given a session_proxy, search `parent` recursively to find the real
+# ShinySession object. If given a ShinySession, simply return it.
+find_ancestor_session <- function(x, depth = 5) {
```

Comment:
Is there any way to know if there are more than 5 nested modules? I doubt it... but maybe we could raise it to 100 or something highly unlikely? (I think 5 is still obtainable. 100 is not.)

Reply:
I'll raise it to 20.